File: Leetcode/268-Missing-Number/MissingNumber.java
public class MissingNumber {

    public int missingNumber(int[] nums) {
        if (nums == null || nums.length == 0) {
            return -1;
        }
        int res = 0;
        for (int i=0; i<nums.length; i++) {
            res = res ^ nums[i] ^ (i+1);
        }
        return res;
    }
}

File: Leetcode/268-Missing-Number/MissingNumber.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var missingNumber = function(nums) {
    if (!nums || nums.length == 0) {
        return -1
    }
    var res = 0
    for (var i=0; i<nums.length; i++) {
        res = res ^ nums[i] ^ (i+1)
    }
    return res
};
